jenkins-builds/production_builds.sh

```diff
@@ -1,35 +1,94 @@
 # Hence one should not use it as CI.
 # The xalt list of modulefiles will be updated only by user jenkins. So this script should only be used by jenkins.
 
+scriptname=`basename $0`
+
+usage() {
+    echo "Usage: $0 [OPTIONS] <list-of-ebfiles>
+    -o OS (required)
+    -a archtecture (required only on daint)
+    -l production list (contains list of eb files)
+    -f force the build for a given package
+    -h help
+    "
+    exit 1;
+}
+
+shortopts="h,o:,a:,l:,f:"
+eval set -- $(getopt -o ${shortopts} -n ${scriptname} -- "$@" 2> /dev/null)
+
+eb_files=()
+while [ $# -ne 0 ]; do
+    case $1 in
+        -h | --help)
+            usage
+            exit 0 ;;
+        -o)
+            shift
+            OS="$1" ;;
+        -a)
+            shift
+            ARCH="$1" ;;
+        -f)
+            shift
+            eb_files+=("$1 -f")
+            ;;
+        -l)
+            shift
+            mapfile -t list < $1
+            for ((i = 0; i < ${#list[@]}; i++)); do
+                eb_files+=("${list[$i]}")
+            done
+            production_file=$1
+            ;;
+        --)
+            ;;
+        *)
+            eb_files+=($1)
+            ;;
+    esac
+    shift
+done
+
 # Retrieve host name (excluding node number)
 if [[ -z $hostName ]] ; then
     hostName=`uname -n | cut -c1-5`
 fi
 
-# expects production file as command line argument
-if [ -z "$1" ]; then
- echo -e "\n Error! Please insert production file on command line! \n"
- exit 1
-else
- filename=$(basename $1)
+if [ "X$OS" == "X" ]; then
+    echo "No OS defined. Please use the option -o to define the OS"
+    exit 1
+fi
+
+if [ "X$ARCH" == "X" ] && [ "Xdaint" == "X$hostName" ]; then
+    echo "No architecture defined. Please use the option -a to define the architecture"
+    exit 1
 fi
 
+#
+# These are additional args to be used based on the system
+#
+eb_args=""
+
 # defines OS VERSION and ARCH parsing the selected production filename
-OS=${filename%-[0-9]*.[0-9]*-[a-z]*}
-ARCH=${filename#$OS-[0-9]*.[0-9]*-}
-echo -e "\n Production file is $1: \n - OS VERSION is $OS \n - ARCH is $ARCH"
+echo -e "\n Production file is: $production_file \n - OS VERSION is '$OS' \n - ARCH is '$ARCH'"
 
 # list of builds
-list=$(cat $1 | grep -v ^#)
-echo -e "\n List of production builds: \n$list"
+echo -e "\n List of production builds with additional options: \n"
+for ((i = 0; i < ${#eb_files[@]}; i++)); do
+    echo ${eb_files[$i]}
+done
 
 # cscs ARCH setup
-module load daint-$ARCH
-echo -e "\n Loading modules: \n - module load daint-$ARCH"
-module rm xalt
+if [ "Xdaint" == "X$hostName" ]; then
+    module load daint-$ARCH
+    echo -e "\n Loading modules: \n - module load daint-$ARCH"
+    module rm xalt
+    eb_args="${eb_args} --modules-header=$APPS/UES/login/daint-${ARCH}.h"
+fi
 
 # EasyBuild setup
 export EASYBUILD_PREFIX=$APPS/UES/jenkins/$OS/$ARCH/easybuild
```

Comment:
I'd really prefer to avoid hardcoding a path here, so that we can use this script in a more flexible way (such as for apps under /apps/common).

My suggestion would be to replace `-o OS` with `-p PREFIX`